### PR TITLE
STYLE: Use f-string in io/parsers

### PR DESCRIPTION
**Pull request by naomi172839**
xref #29547

Fixed many of the strings in pandas/io/parsers.py except for the string at line 1501 as that will require more time to make it function properly.  Please note that this is my first contribution ever so please double check my work!

### CODE REVIEW

**Comment by datapythonista** [comment]
Thanks for the work on this @naomi172839 

If you check your proposed changes in https://github.com/pandas-dev/pandas/pull/30466/files you'll see that there are many unrelated changes (changes in the indentation).

To keep things simple, we don't merge PRs with side changes. And in this case, they are also incorrect based on Python standards.

I guess the changes were done automatically by your editor, or a different version of black than you should have. In any case, please have a look and make sure that you're only proposing the intended changes.

Let me know if you need help.

**Comment by naomi172839** [comment]
I believe that pycharm automatically formats the code.  I will correct the issue and submit a new pull request.

**Comment by datapythonista** [comment]
> I believe that pycharm automatically formats the code. I will correct the issue and submit a new pull request.

I was guessing is the editor. I don't know anything about PyCharm, but you may want to disable whatever is modifying the code automatically.

You don't need to create a new PR. Make the changes locally in your branch, commit them, and push them.

For the next time, you should create a git branch before starting working on something you want to propose. Working on `master` directly means that now you can't work on something in parallel now (may not be a problem right now, but will be in the future).

**Comment by naomi172839** [comment]
@datapythonista  Thank you for the advice! I ran "black" on the file and recommitted it.  Hopefully that is all that I needed to do.  Also thank you for your time!

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: can you remove this blank line please

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: add this back (this will make `flake8` fail, having two lines among top-level functions is required in PEP-8)

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: same

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: the `f` is only needed in the first line, not the last two

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: same, only the line in the middle needs to be an f-string

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: same

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: same

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]:
```suggestion
                        col = col[:-1] + f"{col[-1]}.{cur_count}"
```
the parenthesis seem unnecesary

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]:
```suggestion
                            "Both a converter and dtype were specified "
```

`f` not needed

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]:
```suggestion
                                f"Bool column has NA values in column {c}"
```

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: remove this blank line please

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: `f` not needed

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]:
```suggestion
                    f"Unable to convert column {column} to type {cast_type}"
```

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]:
```suggestion
                            f"Passed header={hr} but only {self.line_pos + 1} lines in "
```

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]:
```suggestion
                    f"{actual_len}"
```
this needs to be a f-string

**Review by datapythonista — COMMENTED** [review]
Thanks for the work on this @naomi172839, really nice clean up.

Added some comments, but looks good

File: pandas/io/parsers.py
Comment (naomi172839) [reply pandas/io/parsers.py]: I was unsure if this was supposed to be 2 lines or just one.  I went with two since you said top level functions require 2 lines.  I am coming from Java so python styling is still strange to me!

Comment (naomi172839) [reply pandas/io/parsers.py]: I was unsure if this was supposed to be 2 lines or just one.  I went with two since you said top level functions require 2 lines.  I am coming from Java so python styling is still strange to me!

Comment (datapythonista) [reply pandas/io/parsers.py]: This is the Python style convention used by (mostly) everybody: https://www.python.org/dev/peps/pep-0008/

Black should take care of it I guess. The main thing is to check `git diff upstream/master` (or the `Files Changed` tab in GitHub, and make sure you don't change things unrelated to what you're addressing.

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: You're adding a blank line now.

Comment (naomi172839) [reply pandas/io/parsers.py]: removed the line.  Hopefully everything is in line with PEP8 now.  Sorry that this created so much work for you!  And thank you for walking me through it.

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: `{icol:s}` is not the same as the original `{icol!s}`. If `icol` is not a string, this will fail, since `s` won't be a valid format, while `!s` calls `str()` and should always work.

File: pandas/io/parsers.py
Comment (datapythonista) [inline pandas/io/parsers.py]: Restore this please.

**Review by datapythonista — COMMENTED** [review]
Looks good, just couple of small things, but almost ready to be merged.

Don't worry about the work on reviewing. Iterating few times is normal and expected when people is starting. Keep contributing as much as you want, we're happy to review.

File: pandas/io/parsers.py
Comment (naomi172839) [reply pandas/io/parsers.py]: was unaware of this difference.  Thanks

**Review by datapythonista — APPROVED** [review]
lgtm, thanks for the work on this @naomi172839, nice PR.

**Comment by naomi172839** [comment]
Thanks again @datapythonista!  Now that everything passes is there anything else that I am supposed to do?

**Comment by datapythonista** [comment]
> Now that everything passes is there anything else that I am supposed to do?

No, now someone else will have a look, and either have comments, or will merge this. We like to have two people from the team reviewing all the changes, that's why I didn't merge this myself (I approved, so next maintainer can merge if they are happy).

**Comment by jreback** [comment]
thanks @naomi172839

**Pull request merged**
